test/interactive_setup_api_integration/fixtures/README.md

```diff
@@ -6,27 +6,86 @@ The Elasticsearch HTTP layer keystore is supposed to mimic the PKCS12 keystore t
 - A PrivateKeyEntry for the CA's key and certificate
 - A TrustedCertificateEntry for the CA's certificate
 
+__IMPORTANT:__ CA keystore (ca.p12) is not checked in intentionally, talk to @elastic/kibana-security if you need it to sign new certificates.
+
 ```bash
-$ES_HOME/bin/elasticsearch-certutil cert \
-  --out $KIBANA_HOME/test/interactive_setup_api_integration/fixtures/elasticsearch.p12 \
-  --ca $KIBANA_HOME/packages/kbn-dev-utils/certs/ca.p12 --ca-pass "castorepass" --pass "storepass" \
-  --dns=localhost --dns=localhost.localdomain --dns=localhost4 --dns=localhost4.localdomain4 \
+ROOT_CA_PATH='packages/kbn-dev-utils/certs/ca.p12'
+ROOT_CA_NAME='root'
+INTERMEDIATE_CA_NAME='intermediate'
+INSTANCE_NAME='elasticsearch'
+
+# Create intermediate CA
+bin/elasticsearch-certutil ca --ca-dn "CN=Elastic Intermediate CA" -days 18250 --out $INTERMEDIATE_CA_NAME.p12 --pass castorepass
+
+# Create instance certificate
+bin/elasticsearch-certutil cert \
+  --ca $INTERMEDIATE_CA_NAME.p12 --ca-pass castorepass --name $INSTANCE_NAME \
+  --dns=localhost --dns=localhost.localdomain \
+  --dns=localhost4 --dns=localhost4.localdomain4 \
   --dns=localhost6 --dns=localhost6.localdomain6 \
-  --ip=127.0.0.1 --ip=0:0:0:0:0:0:0:1
-```
+  --ip=127.0.0.1 --ip=0:0:0:0:0:0:0:1 \
+  -days 18250 --out $INSTANCE_NAME.p12 --pass storepass
 
-Change the alias of the TrustedCertificateEntry so that it won't clash with the CA PrivateKeyEntry
-```bash
+# Convert P12 keystores to PEM files
+openssl pkcs12 -in $ROOT_CA_PATH -out $ROOT_CA_NAME.crt -nokeys -passin pass:castorepass -passout pass:
+openssl pkcs12 -in $ROOT_CA_PATH -nocerts -passin pass:castorepass -passout pass:keypass | openssl rsa -passin pass:keypass -out $ROOT_CA_NAME.key
+
+openssl pkcs12 -in $INTERMEDIATE_CA_NAME.p12 -out $INTERMEDIATE_CA_NAME.crt -nokeys -passin pass:castorepass -passout pass:
+openssl pkcs12 -in $INTERMEDIATE_CA_NAME.p12 -nocerts -passin pass:castorepass -passout pass:keypass | openssl rsa -passin pass:keypass -out $INTERMEDIATE_CA_NAME.key
+
+openssl pkcs12 -in $INSTANCE_NAME.p12 -out $INSTANCE_NAME.crt -clcerts -passin pass:storepass -passout pass:
+openssl pkcs12 -in $INSTANCE_NAME.p12 -nocerts -passin pass:storepass -passout pass:keypass | openssl rsa -passin pass:keypass -out $INSTANCE_NAME.key
+
+# Re-sign intermediate CA
+mkdir -p ./tmp
+openssl x509 -x509toreq -in $INTERMEDIATE_CA_NAME.crt -signkey $INTERMEDIATE_CA_NAME.key -out ./tmp/$INTERMEDIATE_CA_NAME.csr
+dd if=/dev/urandom of=./tmp/rand bs=256 count=1
+touch ./tmp/index.txt
+echo "01" > ./tmp/serial
+cp /System/Library/OpenSSL/openssl.cnf ./tmp/
```

Comment:
__question:__ is there any chance to make this script work on non-Mac machines too? :slightly_smiling_face: Would just `touch ./tmp/openssl.cnf` be enough or we need some other config options from the original `openssl.cnf`? Also from which directory we're supposed to run this script?

To not block the fix, let's discuss this in the follow-up.

Reply:
> Would just `touch ./tmp/openssl.cnf` be enough or we need some other config options from the original `openssl.cnf`?

It's been a long time since I looked at it, but IIRC we do need at least some settings from the default config file to make this work; that file is very large and it wasn't easy to isolate exactly what we needed to copy from, so I copied the whole thing. 

As Thom said, automating cert generation is a high priority :+1: